lib/compiler-singlepass/src/machine.rs

```diff
@@ -20,11 +19,25 @@ pub(crate) struct Machine {
     locals_offset: MachineStackOffset,
 }
 
+/// Returns an u32 that has as 1 bits the ones matching registers passed as parameters
+macro_rules! bitset_of_regs {
+    ($( $r:expr ),*) => {{
+        $( (1u32 << ($r as u32)) )|*
+    }}
+}
+
+// Note: the below asserts are because we currently use u32 for used_gprs and used_xmms
+// Feel free to increase the number in this assert by making them bigger if needed
+#[allow(dead_code)]
```

Comment:
Hm, if this starts with `_`, the allow_dead_code should not be required I think. Maybe it's the problem with how lints are being run? 

https://github.com/near/wasmer/blob/near-main/Makefile#L333

looks iffy to me. I'd expect *just* `-D warnings`. Out of scope for PR obviously

Reply:
I generally dislike `-D warnings` personally. Listing the specific lints allows them to be discussed for their usefulness in isolation, and we’re not at the mercy of whatever clippy developers figure to be a good/bad idea for a specific lint group (naturally there can still be breaking changes but they are much easier to reason about.)

The lint itself is firing on the methods that are being used by these constants, so even if the constant is prefixed with `_` all of the other dead code will remain dead for the purposes of the lint. Marking an use as `#[allow(dead_code)]` also marks everything the marked item uses as actually used which is the desired behaviour here.

Reply:
Ahhhh, I missed that that's clippy, I thought those are just compiler warnings. 

>The lint itself is firing on the methods that are being used by these constants, so even if the constant is prefixed with _ all of the other dead code will remain dead for the purposes of the lint.

Hm, I was sure that `_` prefix makes the compiler consider the symbol as used, but appraently that's just not the case.